### ISSUE TITLE
Signal handling

### ISSUE THREAD

**Issue by lofcek** [unittest-cpp/unittest-cpp#178]
Is there any way how not to handle signals on Linux? It looks that Unittest++ always try to suppress many signals, that appear during running the test. It is not always what we want, especially for SIGSEGV.

I have one teoretical problem - that if some data in the memory are corrupted, next test will be running as undefined behavior.

My practical problem is deadlock. This was caused by crash in test. My test started with a lock a mutex to singleton, after that it crashed (access to null pointer). After that framework tried to fix the signal. Skipped all the destructor (including ~lock_guard) and next test tried to lock the same mutex again.

**Comment by pjohnmeyer** [comment]
@lofcek I don't believe there is an easy way, presently, to disable UnitTest++'s signal trapping behavior.

If you think it would have value, I would welcome a pull request, but I'd like to see some discussion hear about how folks would implement it first.
